pkgs/applications/window-managers/sway/lock-fprintd.nix

```diff
@@ -0,0 +1,43 @@
+{ lib, stdenv, fetchFromGitHub, meson, ninja, pkg-config, scdoc, wayland-scanner
+, wayland, wayland-protocols, libxkbcommon, cairo, gdk-pixbuf, pam
+, dbus, glib, fprintd }:
+
+stdenv.mkDerivation rec {
+  pname = "swaylock-fprintd";
+  version = "unstable-20230130";
+
+  src = fetchFromGitHub {
+    owner = "SL-RU";
+    repo = "swaylock-fprintd";
+    rev = "ffd639a785df0b9f39e9a4d77b7c0d7ba0b8ef79";
+    hash = "sha256-2VklrbolUV00djPt+ngUyU+YMnJLAHhD+CLZD1wH4ww=";
+  };
+
+  postPatch = ''
+    sed -i fingerprint/meson.build \
+      -e 's|/usr/share/dbus-1/interfaces/|'${ fprintd }'/share/dbus-1/interfaces/|'
+  '';
+
+  strictDeps = true;
+  depsBuildBuild = [ pkg-config ];
+  nativeBuildInputs = [ meson ninja pkg-config scdoc wayland-scanner dbus glib ];
+  buildInputs = [ wayland wayland-protocols libxkbcommon cairo gdk-pixbuf pam fprintd ];
+
+  mesonFlags = [
+    "-Dpam=enabled" "-Dgdk-pixbuf=enabled" "-Dman-pages=enabled"
+  ];
+
+  meta = with lib; {
+    description = "Screen locker for Wayland";
+    longDescription = ''
+      swaylock is a screen locking utility for Wayland compositors.
+      Important note: If you don't use the Sway module (programs.sway.enable)
+      you need to set "security.pam.services.swaylock = {};" manually.
```

Comment:
We should probably mention that it is like swaylock but with fingerprint support via fprintd. Or even better: A fork of swaylock with fingerprint support via fprintd.

Maybe something like this (based on the GitHub description?:
> swaylock-fprintd is a fork of swaylock, a screen locking utility for Wayland compositors, with fingerprint support via fprintd. It is compatible with any Wayland compositor which implements the ext-session-lock-v1 Wayland protocol.